### ISSUE TITLE
Implement Drupal data model for Adaptation options

### ISSUE THREAD

**Issue by patrickkaleta** [clarity-h2020/docker-drupal#64]
### Proposed data model
After going through the following documents on adaptation options
- [Adaptation options schema for data package](https://github.com/clarity-h2020/data-package/blob/master/schemas/adaptation-options/adaptation-options.schema.json)
- [current mockups for Adaptation options](https://newrepository.atosresearch.eu/index.php/apps/files/?dir=/CLARITY/WP1%20CO-Creation/T1.3%20Climate%20Services%20Co-Creation/CSIS%20Product%20Mockups&fileid=529) (slide 5 and 20)
- [Adaptation options excel sheet from PLENIVS](https://newrepository.atosresearch.eu/index.php/apps/files/?dir=/CLARITY/WP3%20Science%20Support/T3.4%20Adaptation%20Strategies%20and%20Decision%20Support/Adaptation%20measures%20modelling&fileid=588)

I would propose the following data model to be implemented in the CSIS Drupal system for a new content type "Adaptation option":

- **Title** - can be any string (e.g. Constructions on piles)
- **Id** - string following a predefined format for match-making with data packages (e.g. adaptation-option:constructions-on-piles)
- **Image**
- **Description**
- **Targeted el. at risk** - multiple taxonomy references to [Elements at risk](https://csis.myclimateservice.eu/admin/structure/taxonomy/manage/elements_at_risk/overview)
- **Effects** (or also called "**Variations**" by @maesbri) - multiple references to a new Paragraph "**Effect**", which consists of:
  - **Involved hazard** - single taxonomy reference to [Hazards](https://csis.myclimateservice.eu/admin/structure/taxonomy/manage/hazards/overview)
  - **Change on local effect** - single value from list with values: 0,1,2,3
  - **Change on vulnerability** - single value from list with values: 0,1,2,3
- **Cost new development** - single value from list with values: 0,1,2,3
- **Cost retrofitting** - single value from list with values: 0,1,2,3
- **Cost maintenance** - single value from list with values: 0,1,2,3
- **Co-benefits** - multiple taxonomy references to a new taxonomy "**Co-Benefits**":
  - terms in this taxonomy would consist of the type of co-benefit and its intensity (magnitude), e.g.: air-quality-1, air-quality-2, air-quality-3, biodiversity-1, biodiversity-2,....
- **Local Effects parameters** - multiple taxonomy references to new taxonomy "**Local Effects parameters**":
  - terms in this taxonomy would look like this: albedo, altitude, building shadow,...
<br>
Any feedback, opinions, objections? If not this could be implemented in the CSIS by either me or maybe @DanielRodera (would you have time for that?), since I have currently unfinished business with Twins, Showcases and Report images. 

### Regarding "Effects"
Other possible names for this could be e.g. impact or variation (other suggestions?).
It would also be possible to model this as a new content type instead of a paragraph, but I don't know if this is really necessary and we already have quite a lot of different content types. Main difference between them is that it's not possible to reference (re-use) already existing entities of a paragraph as explained [here](https://drupal.stackexchange.com/questions/206306/how-to-reference-existing-paragraph-entity#answer-206393).

### Regarding "Co-Benefits"
This could also be implemented as a paragraph or content type, which would make it then easier to do filtering such as: show all adapt. opt. with biodiversity value of 2 or higher. Using taxonomies is more suited for making exact filtering like: show all adapt opt. with biodiversity equal to 2. (idk what would be more useful for the user, but implementing it as a taxonomy is in my opinion easier and should therefore be prefered). 

### Regarding "Local Effects parameters"
Everywhere I looked this was always just about whether a parameters is present/relevant for a adaptation option or not, but never were there any actual values for those parameters present. I have no idea how those local effects calculations will look like - is it really enough to just "tell" them that a specific parameter is relevant without providing any actual values for it?

**Comment by maesbri** [comment]
Hi Patrick,

Some comments from my side:

* "effects" instead of "variations" should be fine for me. I think I took the naming from the original excel file as used as basis for creating the json.

* values for the co-benefits intensity: I would propose something like this: air-quality:1, air-quality:2, etc. It would make it easier to parse afterwards (that way we know that after ":" comes the value)

* we should also agree on the taxonomies internal ids for the adaptation-options, local effects parameters and co-benefits catagories (I more or less did it in the schema), which are also used within the data package specification (like the hazards, elements-at-risk, etc.).

**Comment by p-a-s-c-a-l** [comment]
Thanks @patrickkaleta, to me, this looks reasonable.

**Comment by DanielRodera** [comment]
> Any feedback, opinions, objections? If not this could be implemented in the CSIS by either me or maybe @DanielRodera (would you have time for that?), since I have currently unfinished business with Twins, Showcases and Report images.

Thanks @patrickkaleta. Sure, but take into account we have bank holiday on Thursday and Friday (and I took also Monday as vacation). So on my return I will take care of that!

**Comment by patrickkaleta** [comment]
> 
> Thanks @patrickkaleta. Sure, but take into account we have bank holiday on Thursday and Friday (and I took also Monday as vacation). So on my return I will take care of that!

Thanks @DanielRodera ! I agree with the comments made by @maesbri, so please take them into account when creating this data model.

**Comment by patrickkaleta** [comment]
@DanielRodera what's the status on the Adaptation options? You said you could take care of the implementation, but so far I don't see anything in the CSIS. Can you get this done by the end of this week? Otherwise I will take care of it.

**Comment by DanielRodera** [comment]
Hi @patrickkaleta , first of all, sorry for the delay on this task.
I have created 3 new taxonomies following the sources you posted at the beginning:

- **Adaptation options** (with id field that matches with the data-packcage data model id)

- **Co-benefits**

- **Local effects parameters**

Regarding the content types I  have created three new ones: 

- **adaptation_options**: with all the required fields in the description a te beginning except "co-benefits" and "effects" fields wich are included as the following content types 

- **adaptation_options_co-benefits**: with the "co-benefits" (multiple co-benefits taxonomy references) and "intensity" (single value from list with values: 0,1,2,3) fields 

- **adaptation_options_effects**: with "Involved hazard" (single taxonomy reference), "change on local effect" (single value from list with values: 0,1,2,3) and "Change on vulnerability" (single value from list with values: 0,1,2,3). I created a content type instead a paragrah as you suggested

I hope is that you was expecting, if you need to change something please let me know.

**Comment by patrickkaleta** [comment]
> Hi @patrickkaleta , first of all, sorry for the delay on this task.
> I have created 3 new taxonomies following the sources you posted at the beginning:
> 
> * **Adaptation options** (with id field that matches with the data-packcage data model id)
> * **Co-benefits**
> * **Local effects parameters**
> 
> Regarding the content types I have created three new ones:
> 
> * **adaptation_options**: with all the required fields in the description a te beginning except "co-benefits" and "effects" fields wich are included as the following content types
> * **adaptation_options_co-benefits**: with the "co-benefits" (multiple co-benefits taxonomy references) and "intensity" (single value from list with values: 0,1,2,3) fields
> * **adaptation_options_effects**: with "Involved hazard" (single taxonomy reference), "change on local effect" (single value from list with values: 0,1,2,3) and "Change on vulnerability" (single value from list with values: 0,1,2,3). I created a content type instead a paragrah as you suggested
> 
> I hope is that you was expecting, if you need to change something please let me know.

After having a closer look at this and creating a couple of adaptation options based on the Excel sheet provided by PLENIVS, I think that the **content type** [adaptation_options_co-benefits](https://csis.myclimateservice.eu/admin/structure/types/manage/adaptation_options_co_benefits/fields) and the **taxonomy** [Adaptation Options](https://csis.myclimateservice.eu/admin/structure/taxonomy/manage/adaptationoptions/overview) can be removed, since they are both unnecessary and not used. I will remove them after ECCA.

@DanielRodera I'm sorry if my description in the beginning was a little misleading. When I wrote that something "could also be implemented as ...", I meant that as "instead of" and not as "additionally to". It usually doesn't make much sense to implement something in Drupal as a content type (which are used to describe/store actual objects) and as a taxonomy as well (which are used to classify/categorize content).

Also, I noticed a strange new content type called [ad](https://csis.myclimateservice.eu/admin/structure/types/manage/ad/fields). Did you maybe create that one by accident while working on the Adaptation option's content type? IMO this one can be removed as well, since it has no purpose.

**Comment by DanielRodera** [comment]
> @DanielRodera I'm sorry if my description in the beginning was a little misleading. When I wrote that something "could also be implemented as ...", I meant that as "instead of" and not as "additionally to". It usually doesn't make much sense to implement something in Drupal as a content type (which are used to describe/store actual objects) and as a taxonomy as well (which are used to classify/categorize content).

No problem @patrickkaleta. I am no so much experienced in Drupal so I didnt realized about that, thank you for the advices.

Regarding the "ad" content type, maybe I created this one by mistake, so can be removed.

Thank you!

**Comment by patrickkaleta** [comment]
I removed the unnecessary content type and taxonomy. Also, I added the remaining adaptation options from the PLENIVS excel sheet. Since that excel sheet is not complete, some adaptation options have missing attributes. Better images and descriptions are needed for all adaptation options.

IMO we can close this issue now and discuss further improvements for the Adaptation options in separate issues.
